parser/charset/encoding_gbk.go

```diff
@@ -91,3 +92,41 @@ var GBKCase = unicode.SpecialCase{
 	unicode.CaseRange{Lo: 0x01DC, Hi: 0x01DC, Delta: [unicode.MaxCase]rune{0, 0, 0}},
 	unicode.CaseRange{Lo: 0x216A, Hi: 0x216B, Delta: [unicode.MaxCase]rune{0, 0, 0}},
 }
+
+// customGBK is a simplifiedchinese.GBK wrapper
+type customGBK struct{}
+
+// NewDecoder returns simplifiedchinese.GBK.NewDecoder()
+func (c customGBK) NewDecoder() *encoding.Decoder {
+	return &encoding.Decoder{
+		Transformer: customGBKDecoder{
+			gbkDecoder: simplifiedchinese.GBK.NewDecoder(),
+		},
+	}
+}
+
+// NewEncoder returns simplifiedchinese.GBK.NewEncoder()
+func (c customGBK) NewEncoder() *encoding.Encoder {
+	return simplifiedchinese.GBK.NewEncoder()
+}
+
+type customGBKDecoder struct {
+	gbkDecoder *encoding.Decoder
+}
+
+// Transform special treatment for 0x80
+// see https://github.com/pingcap/tidb/issues/30581 get details
+func (c customGBKDecoder) Transform(dst, src []byte, atEOF bool) (nDst, nSrc int, err error) {
+	if len(src) == 0 {
+		return 0, 0, nil
+	}
+	if src[0] == 0x80 {
```

Comment:
Please add a test like 
```
set names gbk;
create table t3(a char(10) charset gbk);
insert into t3 values('一a');
```

Reply:
The similar test case:
https://github.com/pingcap/tidb/pull/30757/files#diff-6eaacb6a6fa7a8b6851fa3bc9917a3cfc6a7a3e191069b6d59ae5385b9beb6faR6426

Reply:
It doesn't have `insert` statement.

Reply:
Updated

Reply:
Seems the previous problem is fixed, but still have some other problems.
```
mysql> insert into t3 values('一a');
Query OK, 1 row affected, 1 warning (0.01 sec)

mysql> show warnings;
+---------+------+--------------------------------------+
| Level   | Code | Message                              |
+---------+------+--------------------------------------+
| Warning | 1300 | Invalid gbk character string: '8061' |
+---------+------+--------------------------------------+
1 row in set (0.00 sec)
```